File: 2018/HW-5/p3/p3.go
/*
 *
 * In The Name of God
 *
 * +===============================================
 * | Author:        Parham Alvani <parham.alvani@gmail.com>
 * |
 * | Creation Date: 23-11-2018
 * |
 * | File Name:     p3.go
 * +===============================================
 */

package main

import "fmt"

func main() {
	var n int

	fmt.Scanf("%d", &n)

	i := 1
	a := 1
	b := 2
	for i < n {
		if i%2 == 0 {
			b = a + b
		} else {
			a = a + b
		}
		i++
	}
	if n%2 == 0 {
		fmt.Println(b)
	} else {
		fmt.Println(a)
	}
}

File: Fall-2016/HW-4/p3.c
/*
 * In The Name Of God
 * ========================================
 * [] File Name : p3.c
 *
 * [] Creation Date : 15-11-2016
 *
 * [] Created By : Parham Alvani (parham.alvani@gmail.com)
 * =======================================
*/
/*
 * Copyright (c) 2016 Parham Alvani.
*/
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char *args[])
{
	int n;
	int i;
	int m = 0;

	scanf("%d", &n);

	srand(time(NULL));
	
	for (i = 0; i < n; i++) {
		long long x, y;
		x = rand();
		y = rand();
		if (x * x + y * y <= (long long) RAND_MAX * RAND_MAX)
			m++;
	}
	
	double p = (double) m / n;
	printf("%.4lf\n", 4 * p);
}
